doc/cli_auto_generation/cli_auto_generation.md

```diff
@@ -0,0 +1,653 @@
+                leaf default_bgp_status {
+                    type enumeration {
+                        enum up;
+                        enum down;
+                    }
+                    default up;
+                }
+                leaf hostname {
+                    type string {
+                        length 1..255;
+                    }
+                }
+                leaf platform {
+                    type string {
+                        length 1..255;
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+###### `config` command
+```
+admin@sonic:~$ config device-metadata localhost hwsku "ACS-MSN2100"
+admin@sonic:~$ config device-metadata localhost default-bgp-status up
+admin@sonic:~$ config device-metadata localhost hostname "r-sonic-switch"
+admin@sonic:~$ config device-metadata localhost platform "x86_64-mlnx_msn2100-r0"
+```
+
+__The `show` command produces named columns. Each column name is an uppercase of `leaf` name from the YANG model, if `leaf` contain `-` or `_` those will be trimmed:__
+
+###### show command
+```
+admin@sonic:~$ show device-metadata localhost
+
+HWSKU        DEFAULT BGP STATUS  HOSTNAME        PLATFORM
+-----        ------------------  --------        --------
+ACS-MSN2100  UP                  r-sonic-switch  x86_64-mlnx_msn2100-r0
+```
+
+###### Config DB schema
+```
+{
+	"DEVICE_METADATA": {
+		"locahost": {
+			"hwsku": "ACS-MSN2100",
+			"default_bgp_status": "up",
+			"hostname": "r-sonic-switch",
+			"platform": "x86_64-mlnx_msn2100-r0"
+		}
+	}
+}
+```
+__3. For every `list` element will be generated `add/del/update` commands:__
+
+For instance let's take a PART of existing [sonic-vlan.yang](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-vlan.yang)
+
+###### sonic-vlan YANG model
+```yang
+module sonic-vlan {
+	// ...
+	container sonic-vlan {
+		// ...
+		container VLAN {
+			list VLAN_LIST {
+				key "name";
+				leaf name {
+					type string {
+						pattern 'Vlan([0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4])';
+					}
+				}
+				leaf vlanid {
+					type uint16 {
+						range 1..4094;
+					}
+				}
+				leaf mtu {
+					type uint16 {
+						range 1..9216;
+					}
+				}
+				leaf admin_status {
+					type stypes:admin_status;
+				}
+			}
+		}
+	}
+}
+```
+
+In the case of below, `Vlan11` - is a positional argument and `key` for the `list`.
+`vlanid`, `mtu`, `admin-status` - are not-positional arguments, and to give them the next style MUST be used (check `config command`)
+__This style "--arg" is NOT RELATED to the Linux CLI optional parameter style__
+
+###### `config` command
+```
+admin@sonic:~$ config vlan add Vlan11 --vlanid 11 --mtu 128 --admin-status up
+admin@sonic:~$ config vlan del Vlan11
+```
+
+The `update` command will be generated but not always could be executed. The user will get an error message from the YANG validator if the `update command` is not supported.
```

Comment:
There are a few references to the `update` subcommand, but no examples. Would it be possible to provide an example of the `update` subcommand, if it is supported?

Reply:
fixed